components/SideBar.tsx

```diff
@@ -0,0 +1,86 @@
+import React, { SVGProps } from 'react';
+import { useGlobalContext } from './context';
+import Link from 'next/link';
+import {
+  LogoutIcon,
+  EditIcon,
+  LocationIcon,
+  DashboardIcon,
+  ExitIcon,
+} from './icons/sidebar_icons';
+
+const Sidebar = () => {
+  const { isSidebarOpen, closeSidebar } = useGlobalContext();
+
+  return (
+    <div
+      className={`transition-all  duration-500  fixed top-0 ${
+        isSidebarOpen ? 'left-0' : '-left-64'
+      }`}
+    >
+      <div className="flex h-screen overflow-y-auto flex-col bg-ocf-black w-64 px-4 py-8 min-h-screen relative">
+        <button
+          onClick={closeSidebar}
+          className="absolute top-1 right-1 text-white w-8 h-8 rounded-full flex items-center justify-center focus:outline-none ml-6"
+        >
+          <ExitIcon />
+        </button>
+        <div className="text-xs	flex flex-col mt-6 justify-between flex-1">
+          <Button
+            href="/dashboard"
+            name="Dashboard"
+            svg={<DashboardIcon />}
+            className="mx-4 font-medium flex-1 align-center text-amber"
```

Comment:
since these `className` attributes all look the same with the exception of text-color (and given that there are only two different text colors) it might be better to move `className="mx-4 font-medium flex-1 align-center"` into the `Button` component and add a prop to determine which text color to use